### PR TITLE
KYLIN-5023 Support cluster deployMode for Standalone

### PR DESCRIPTION
**Pull request by hit-lacus**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

[Design doc](https://www.notion.so/xxyu/Dev-Diary-KYLIN-5023-Kylin-4-deployMode-Cluster-80c269b757a74f9892551701e252ceec)

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [x] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I will prepare another pr against the `document` branch
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hit-lacus** [comment]
## Design 

[ design ](https://www.notion.so/xxyu/Dev-Diary-KYLIN-5023-Kylin-4-deployMode-Cluster-80c269b757a74f9892551701e252ceec)

## Configuration

Enable standaone cluster mode by following configuration.

```
kylin.engine.spark.standalone.master.httpUrl=http://cdh-master:10031
kylin.engine.spark-conf.spark.submit.deployMode=cluster
kylin.engine.spark-conf.spark.shuffle.service.enabled=false
kylin.engine.spark-conf.spark.master=spark://cdh-master:10030
```

**Comment by hit-lacus** [comment]
## Test Case

#### Kill the Driver while the spark application is running
It is clear that when Driver status is not the same as the spark application.
image

File: kylin-spark-project/kylin-spark-engine/src/main/scala/org/apache/spark/deploy/StandaloneAppClient.scala [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+
+  def parseApplicationState(responseStr: String): Unit = {
+    val curr = System.currentTimeMillis()
+
+    var respJson = Map.empty[String, Any]
+    val tree = parseFull(responseStr)
+    respJson = tree match {
+      case Some(map: Map[String, Any]) => map
+    }
```
Comment (zhengshengjun) [inline kylin-spark-project/kylin-spark-engine/src/main/scala/org/apache/spark/deploy/StandaloneAppClient.scala:92]: maybe simplify to: 

val respJson = parseFull(responseStr).get.asInstanceOf[Map[String, Any]]

**Comment by xiacongling** [comment]
Hi, @hit-lacus. Similar with https://github.com/apache/kylin/pull/1682, compilation failed for Spark3 compatibility. To resolve the problem, L495 of KylinExpressions may be changed like the following:

https://github.com/apache/kylin/pull/1680/files#diff-eabd8d352aa09399e10ce9d87d3142f314006e5f43980911d98bc61a42dd50eaL495-R495

**Comment by hit-lacus** [comment]
> Hi, @hit-lacus. Similar with #1682, compilation failed for Spark3 compatibility. To resolve the problem, L495 of KylinExpressions may be changed like the following:
> 
> https://github.com/apache/kylin/pull/1680/files#diff-eabd8d352aa09399e10ce9d87d3142f314006e5f43980911d98bc61a42dd50eaL495-R495

Thanks for let me know, thx.

**Pull request merged**